File: backend/src/main/java/com/ieum/admin/notice/domain/AdminNotice.java
package com.ieum.admin.notice.domain;

import lombok.*;

import java.time.LocalDateTime;

/**
 * 관리자용 공지사항 도메인 모델 (순수 자바 객체)
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminNotice {

    private Long id;
    private String title;
    private String content;
    private String summary;

    @Builder.Default
    private Integer viewCount = 0;

    @Builder.Default
    private Boolean isPinned = false;

    @Builder.Default
    private Boolean isPopup = false;

    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/admin-notice.ts
/**
 * 관리자 공지사항 관리 타입 정의
 */

/** 공지사항 아이템 */
export interface AdminNoticeItem {
  id: number;
  title: string;
  content: string;
  summary?: string;
  viewCount: number;
  isPinned: boolean;
  isPopup: boolean;
  isPushed: boolean;
  startDate?: string;
  endDate?: string;
  createdAt: string;
  updatedAt: string;
}

/** 공지사항 목록 응답 (Spring Data Page) */
export interface AdminNoticeListResponse {
  content: AdminNoticeItem[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}
